### PR TITLE
Shared `ConditionalCompletionSplitting` implementation

### PR DESCRIPTION
**Pull request by hvitved**
Ideally, I would have liked for `SplitImplementations::ConditionalCompletionSplitting` to be parameterized over `ConditionalCompletionSplittingInput` instead of having the outer `MakeWithSplitting` be parameterized over it, but then it would not be possible to extend the abstract `SplitImpl` class.

### CODE REVIEW

**Review by MathiasVP — APPROVED** [review]
LGTM, but let's wait for DCA to come back up before we merge!

**Comment by hvitved** [comment]
DCA looks good, so merging.

**Pull request merged**
